### PR TITLE
ROUTE_HOSTNAME is not mandatory but has a default.

### PR DESCRIPTION
**Pull request by rhuss**
Short explanation about ROUTE_HOSTNAME's default ipaas.127.0.0.1.nip.io added.

### CODE REVIEW

**Comment by jimmidyson** [comment]
I think perhaps we should not default this at all as that default works for me, but not in general case I'd suggest. WDYT?

**Comment by rhuss** [comment]
Well,  tbh, I'm not a big fan of default values anyway ;-) 

By coincidence, I just thought a bit about default in a more general context, which I'm happy to share (its more about defaults within some sort application :)

> #### About default values
>
> Default values are not always a good idea, they might be even an anti-pattern. The reason for this is that _changing_ default values afterwards is a difficult task in general. 
>
> First, changing default values means to change them within the code which requires a rebuild.  
>
> Second, people relying on defaults (either by accident or consciously) will be always be surprised when a default value changes. 
>
> This is has to be communicated and the user of such an application has probably to change the calling code as well.  Changes in default values however might make often sense, also because it is so damned hard to get default values right from the very beginning.  But this has to be considered as a _major change_ and if semantic versioning is in use, such a change justifies a bump in the major version number.  If unsatisfied with an existing default value it is often better to remove the default altogether and throw an error if no value is provided.  This will at least break the application early and prominently instead of doing silently something different and unexpected. Considering all these issues it is often the best solution to _avoid default values_ from the very beginning. Database connection parameters are good candidates for not providing default values as they highly depend on the environment and can often not be reliably predicted. 
Also if no default values are used, then the configuration information which has to be provided from the outside can serve nicely as documentation, too.

**Comment by jimmidyson** [comment]
Without defaults the template becomes unwieldy. I am a fan of defaults to aid usability, as long as defaults are easy to override, as they are here. Plus this is a one-time deployment thing rather than a runtime default. From these template defaults explicit runtime configuration is generated so it's kind of a win-win: easy usability and explicit, non-default reliant, runtime configuration.

**Comment by rhuss** [comment]
Indeed, depends highly on the context. My experience is more about changing defaults in tools like Jolokia (like the default response content-type which currently plain/text), which is hard in retrospective without breaking lot of stuff. Still in this case a default makes sense as a 'good' default can be set and simplifies life. Just hard to change.

For a one-time shot where a default fits 90% of all use cases I completely agree with you. 

It's not black or white.

Back to the PR: If the default route is a good fit and works for most (which I suppose it is as nip.io was invented for this use case), I would keep it.

**Comment by jimmidyson** [comment]
The problem is the 127.0.0.1 actually... that causes pods to point to localhost when they should be pointing out to externally resolvable hostnames.

**Comment by rhuss** [comment]
ah, you mean e.g. when running on minishift where you should use `minishift ip` ?

**Comment by jimmidyson** [comment]
Even with straight `oc cluster up` same applies: route hostname must resolve to external address rather than 127.0.0.1 which it currently does. It just doesn't work any more, since keycloak was introduced as we have multiple services to communicate with.

**Comment by rhuss** [comment]
Then I would leave out the default but provide some examples for important use cases in the README.md

**Comment by jimmidyson** [comment]
Yeah my thinking too.

**Comment by rhuss** [comment]
Ok, let me try to update it.

**Comment by rhuss** [comment]
@jimmidyson could you have a quick look that I'm not talking nonsense in the README ? (and whether the template fixes are correct)

File: README.md [hunk 1]
```diff
@@ -7,12 +7,16 @@ Run the following commands:
...
 $ oc create -f https://raw.githubusercontent.com/redhat-ipaas/openshift-templates/master/redhat-ipaas.yml
-$ oc new-app redhat-ipaas -p ROUTE_HOSTNAME=<EXTERNAL_HOSTNAME>
+$ oc new-app redhat-ipaas -p ROUTE_HOSTNAME=<external hostname>
 ```
 
-Replace `EXTERNAL_HOSTNAME` with a value that will resolve to the address of the OpenShift router.
+Replace _<external hostname>_ with a value that will resolve to the address of the OpenShift router.
 
-Once all pods are started up, you should be able to access the iPaaS at `https://<EXTERNAL_HOSTNAME>/`.
+You should choose for _<external hostname>_ a address which is routable on your system and leads to your OpenShift router. This can be done by modifying your address resolution configuration (e.g. by editing `/etc/hosts`) or by using an external DNS resolving service like xip.io or nip.io without touching your system's configuration (that's the easisest way).
```
Comment (jimmidyson) [inline README.md]: can't use etc hosts because it needs to also resolve inside pods

**Comment by jimmidyson** [comment]
Thanks!

**Pull request merged**